lib/cext/preprocess.rb

```diff
@@ -26,10 +26,18 @@
 		case Qfalse_int_const : mybool = 0; break;
 EOF
 
-PATCHED_FILES = {'xml_node_set.c'      => {:match =>  /[[:blank:]]*?switch\s*?\(.*?Qnil:/m, 
-                                           :replacement => XML_NODE_SET_PATCH},
-                 'pg_binary_encoder.c' => {:match => /[[:blank:]]*?switch\s*?\(.*?Qfalse\s*?:.*?break;/m,
-                                           :replacement => PG_BINARY_ENCODER_PATCH}}
+PATCHED_FILES = {'xml_node_set.c'      => [{:match =>  /[[:blank:]]*?switch\s*?\(.*?Qnil:/m, 
+                                           :replacement => XML_NODE_SET_PATCH}],
+                 'pg_binary_encoder.c' => [{:match => /[[:blank:]]*?switch\s*?\(.*?Qfalse\s*?:.*?break;/m,
+                                           :replacement => PG_BINARY_ENCODER_PATCH}],
+                 'bytebuffer.c'        => [{:match => /(static VALUE .*?) = Qnil;/,
```

Comment:
For this cases, Sulong could actually support it as globals are stored in managed memory, but the C compiler doesn't accept a non constant-literal for initialization.
One way to rewrite it might be to use:
```c
static VALUE var; // no initializer

__attribute__((constructor))
static void initialize_var() {
    var = Qnil;
}
```
and that would actually read `nil` out of it, not `null`.
It could also be moved to the `Init_` function but might be harder to achieve that.
Sometimes the global is overwritten anyway before being read, in which this doesn't really matter much.